test/hotspot/jtreg/gc/g1/TestLargePageUseForAuxMemory.java

```diff
@@ -49,9 +49,25 @@
     static long smallPageSize;
     static long allocGranularity;
 
+    static boolean checkLargePagesEnabled(OutputAnalyzer output) {
+        String lp = output.firstMatch("Large Page Support: (\\w*)", 1);
+        return lp != null && lp.equals("Enabled");
+    }
+
     static void checkSize(OutputAnalyzer output, long expectedSize, String pattern) {
-        String pageSizeStr = output.firstMatch(pattern, 1);
+        // First check if there is a large page failure associated with
+        // the data structure being checked.
```

Comment:
Are you thinking something like this @albertnetymk?
```suggestion
        // First check if there is a large page failure associated with
        // the data structure being checked. In case of a large page 
        // allocation failure the output will include logs like this for 
        // the affected data structure:
        // [0.048s][debug][gc,heap,coops] Reserve regular memory without large pages
        // [0.048s][info ][pagesize     ] Next Bitmap: ... page_size=2M ...
```

Reply:
Yes, and also for `checkLargePagesEnabled`. It's not obvious to me why we parse the output in those two places, one looking for the failure mode, and the other looking for the success mode. That's why I asked for an sample of the "expected" log output.

Reply:
I think the check if large pages are enable is pretty straight forward. We should never expect large page sizes in the output unless large pages are enabled. I do however agree that this check is a bit clunky. Would it help to extract it to a separate function? Something like `largePagesAllocationFailure(pattern)`, I could also change the name of the function above to just be `largePagesEnabled()` then the code reads even better?

Reply:
I overlooked the fact that `largePagesAllocationFailure` is pattern/data structure specific. I am happy with the current patch. Thank you.